Signal/src/Profiles/OWSProfileManager.m

```diff
@@ -404,11 +415,164 @@ - (void)uploadAvatarToService:(NSData *)data
         // TODO:
         NSString *avatarUrl = @"avatarUrl";
         NSData *avatarDigest = [@"avatarDigest" dataUsingEncoding:NSUTF8StringEncoding];
-        if (YES) {
-            successBlock(avatarUrl, avatarDigest);
-            return;
-        }
-        failureBlock();
+
+        // See: https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-UsingHTTPPOST.html
+        TSProfileAvatarUploadFormRequest *formRequest = [TSProfileAvatarUploadFormRequest new];
```

Comment:
https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-UsingHTTPPOST.html

I think the key lies in here somewhere:

https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-HTTPPOSTForms.html